### PR TITLE
257 checkbox component

### PR DESCRIPTION
**Pull request by zlrkw11**
closes

### CODE REVIEW

**Comment by zlrkw11** [comment]
image

File: client/src/components/generic/Checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  variant?: checkboxVariants
+  checked: boolean
+  disabled: boolean
+}
+
+type props = ICheckboxProps & React.InputHTMLAttributes<HTMLInputElement>
+
+const Checkbox = ({ children, ...props }: props) => {
+  return (
+    <div {...props}>
```
Comment (bcho892) [inline client/src/components/generic/Checkbox/Checkbox.tsx:14]: the props should be spread on the `input`

File: client/src/components/generic/Checkbox/Checkbox.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+type checkboxVariants = "default" | "checked" | "disabled"
```
Comment (bcho892) [inline client/src/components/generic/Checkbox/Checkbox.tsx:1]: nit: disabled should already be part of `React.InputHTMLAttributes<HTMLInputElement>`

**Review by bcho892 — CHANGES_REQUESTED** [review]
image
The colours don't seem to quite match?

**Comment by bcho892** [comment]
Might have to refer to figma again

File: client/src/components/generic/Checkbox/Checkbox.tsx (hunk 1 above)
Comment (bcho892) [reply client/src/components/generic/Checkbox/Checkbox.tsx:10]: we need to put this on the input component

**Review by bcho892 — CHANGES_REQUESTED** [review]

**Pull request merged**